### PR TITLE
micro_http: use case-insensitive match for header names

### PR DESCRIPTION
**Pull request by bpot**
## Reason for This PR

HTTP header matching should be case-insensitive. In particular the `hyper` http client appears to use lower case headers which causes an `Empty Put Request` (b/c `content-length` header not found) error.

## Description of Changes

Do a case insensitive comparison.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided.
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.
- [x] Either no new `unsafe` code has been added, or, the newly added `unsafe`
      code is unavoidable and properly documented.

### CODE REVIEW

File: src/micro_http/src/common/headers.rs [hunk 1]
```diff
@@ -32,13 +32,14 @@ impl Header {
     }
 
     fn try_from(string: &[u8]) -> Result<Self, RequestError> {
-        if let Ok(utf8_string) = String::from_utf8(string.to_vec()) {
+        if let Ok(mut utf8_string) = String::from_utf8(string.to_vec()) {
+            utf8_string.make_ascii_lowercase();
```
Comment (KarthikNedunchezhiyan) [inline src/micro_http/src/common/headers.rs]: What do you think about trimming the field name 🙂?

**Comment by KarthikNedunchezhiyan** [comment]
@bpot what do you think about trimming the field name 🙂?

**Comment by bpot** [comment]
@KarthikNedunchezhiyan it looks like the `trim` call is still in there. I could be misunderstanding your comment though!

**Comment by KarthikNedunchezhiyan** [comment]
@bpot yes you are right, i thought it was removed 🙂.

**Pull request merged**